fix: update docs about partitioning.

billy-the-fish opened the pull request.

billy-the-fish committed:
fix: update docs about partitioning.

erimatnor commented on api/add_dimension.md:
```diff
@@ -11,79 +11,52 @@ api:
+
+  Multiple tablespaces only supports concurrent queries.
+
+When using hash partitions, best practice is to use 1 hash partition per disk.
```
I am not sure we have a best practice for this. But I guess it is reasonably to have _at least_ 1 partition per disk. Personally, I would probably say it is better to have number of partitions set to a multiple of number of disks, e.g., the number of partitions P=N*Pd where N is the number of disks and Pd is the number of partitions per disk. Then you could also add more disks later and move partitions to the new disk from other disks. But maybe this is too much information for this section?

billy-the-fish replied:
Updated.

erimatnor commented on api/add_dimension.md:
````diff
@@ -119,7 +92,7 @@ partitioning on column `time`, then add an additional partition key on
 ```sql
 SELECT create_hypertable('conditions', by_range('time'));
-SELECT add_dimension('conditions', by_hash('location', 4));
+SELECT add_dimension('conditions', by_range('time'), by_hash('location', 4));
````
I don't think this change is correct. Looks like you are confusing `create_hypertable` and `add_dimension`.

`create_hypertable` requires adding at least one range dimension as the "primary" partitioning dimension. However, `add_dimension` is called _after_ `create_hypertable` to add additional dimensions, typically a space/hash dimension after the hypertable exists  (including its primary range/time dimension).

billy-the-fish replied:
Updated.

erimatnor commented on api/add_dimension.md:
````diff
@@ -133,13 +106,13 @@ partitionining on `device_id`.
 ```sql
 SELECT create_hypertable('conditions', by_range('time'));
-SELECT add_dimension('conditions', , by_hash('location', 2));
+SELECT add_dimension('conditions', by_range('time'), by_hash('location', 2));
````
Same here as above. And, the changes a few lines below.

billy-the-fish replied:
Updated.

erimatnor commented on api/dimension_info.md:
```diff
@@ -1,41 +1,36 @@
-dimension builders to specify the dimensions to partition a
-hypertable on.
+You call [`create_hypertable`][create_hypertable] and [`add_dimension`][add_dimension] to specify the dimensions to 
+partition a hypertable on. TimescaleDB supports partitioning [`by_range`][by-range] and [`by_hash`][by-hash]. You can 
+partition `by_range` on it's own. However, because `by_hash` makes a fixed number of partitions which can grow 
+very large, best practice is to set `by_range` when you use `by_hash`. 
```
Here I suggest we say that hypertables must always have a primary range/time dimension, followed by an arbitrary number of additional dimensions that can be either range or hash, but typically just one hash.

billy-the-fish replied:
Done

billy-the-fish committed:
fix: add note.

billy-the-fish committed:
fix: update on review.

billy-the-fish committed:
Merge branch 'latest' into add-create-hypertable-link-for-schema-management

billy-the-fish committed:
Merge branch 'latest' into add-create-hypertable-link-for-schema-management

billy-the-fish committed:
Merge branch 'latest' into add-create-hypertable-link-for-schema-management

billy-the-fish committed:
Merge branch 'latest' into add-create-hypertable-link-for-schema-management

billy-the-fish committed:
Merge branch 'latest' into add-create-hypertable-link-for-schema-management

erimatnor commented on api/dimension_info.md:
```diff
@@ -1,41 +1,39 @@
 data type. Look in the example section below.
-</Highlight>
+
+Dimension builders were introduced in TimescaleDB 2.13.
```
I am confused by the section on "dimension builders". I know it was in the docs before, but it seems the concept is never really introduced. What is a dimension builder and why do I care about it?

My understanding is that it is just a set of functions you use together with create_hypertable() and add_dimension(), but those pages don't even mention the concept or make any reference to this section although the "builder" functions are used in those sections.  

I guess am wondering why we need a separate section on dimension builders when it is not mentioned elsewhere?

If we are sticking with this concept of dimension builders, I think this section needs a better introduction to what they are and why they are useful, and when to use them.

But, if it was up to me, I would just take what is useful here and merge it with the sections on create_hypertable and add_dimension. I would probably skip the concept of "dimension builder" and just explain how to use the functions by_range() and by_hash()

billy-the-fish replied:
Thanks for this comment, it really helps. I have had another look over this, and from what I understand this is talking about the _timescaledb_internal.dimension_info type returned when you call create_hypertable or add_dimension. How about as you suggest I add any useful information to those functions, and move the rest of the info to https://docs.timescale.com/api/latest/informational-views/ as a dimension_info object? 

My head hurts.

erimatnor commented on api/add_dimension.md:
```diff
@@ -11,79 +11,62 @@ api:
+  multiple tablespaces to a *single* hypertable. However, although under the hood, a hypertable's
+  chunks are spread across the tablespaces associated with that hypertable.
+
+  Multiple tablespaces only supports concurrent queries.
```
This isn't correct. A single query should be able to read from multiple tablespaces/disks and also run a parallel query across those disks.

erimatnor commented on api/add_dimension.md:
```diff
@@ -11,79 +11,62 @@ api:
+- **RAID**: use a RAID setup across multiple physical disks, and expose a single logical disk to the hypertable.
+  That is, using a single tablespace.
+
+  Best practice is to use RAID when possible. This is because it supports the concurrent and single
+  scenarios and *no spatial partitioning is required*.
```
This was probably there before and I am not sure where it comes from, but this seems to imply that using multiple tablespaces/disks requires spatial partitioning. This is not the case (and I don't think it has ever been the case). You can spread chunks across multiple disks with only time partitioning, so I am not sure exactly why "no spatial partitioning is required" is mentioned since it applies also the "Multiple tablespaces".

I would recommend just removing this last bit.

If we want to mention that RAID is "best practice" (and, honestly, I am not exactly sure why it is best practice), I would motivate it primarily with the fact that you need not manually manage tablespaces in the DB when using RAID. Still, you do need the complexity of a RAID setup instead.

erimatnor commented on api/add_dimension.md:
```diff
@@ -11,79 +11,62 @@ api:
+
+  Multiple tablespaces only supports concurrent queries.
+
+When using hash partitions, best practice is to have at least one hash partition per disk.
+
+set number of partitions to a multiple of number of disks. For example, the number of 
+partitions P=N*Pd where N is the number of disks and Pd is the number of partitions per 
+disk. This enables you to add more disks later and move partitions to the new disk from other disks.
```
```suggestion
When using multiple tablespaces, a best practice is to also add a second hash-partitioned dimension to your hypertable and to have at least one hash partition per disk. While a single time dimension would also work, it would mean that the first chunk is written to one tablespace, the second to another, and so on, and thus would parallelize only if a query's time range exceeds a single chunk.

When adding a hash partitioned dimension, set the number of partitions to a multiple of number of disks. For example, the number of 
partitions P=N*Pd where N is the number of disks and Pd is the number of partitions per 
disk. This enables you to add more disks later and move partitions to the new disk from other disks.
```

billy-the-fish committed:
Update api/add_dimension.md

Co-authored-by: Erik Nordström <819732+erimatnor@users.noreply.github.com>
Signed-off-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
fix: revert structure. (#3471)

* fix: revert structure.

billy-the-fish committed:
chore: updates on review.

esclymeTS committed:
Update changelog.md (#3481)

* Update changelog.md

---------

Signed-off-by: Eric Clymer <eric@timescale.com>
Signed-off-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
chore: update explanation for modes in Timescale Console. (#3474)

* chore: update explanation for modes in Timescale Console.

Co-authored-by: atovpeko <114177030+atovpeko@users.noreply.github.com>
Co-authored-by: Rahil Sondhi <rahilsondhi@gmail.com>

billy-the-fish committed:
chore: Update hypertables API ref.

billy-the-fish committed:
chore: more messing about.

billy-the-fish committed:
chore: update docs with info about multiple CIDR blocks and changes t… (#3479)

* chore: update docs with info about multiple CIDR blocks, changes to pricing, and revalidate the procedure.

atovpeko committed:
Update PostgreSQL version (#3477)

Co-authored-by: Iain Cox <iain@timescale.com>

atovpeko committed:
foreign key constraint update (#3492)

Co-authored-by: Iain Cox <iain@timescale.com>

VineethReddy02 committed:
Remove live-migration reference to docker image URL (#3493)

* Remove reference to docker website

The live-migration documentation serves as our primary resource, with the first line highlighting live-migration.

Additionally, since there's no specific content for live-migration on Docker Hub, we should avoid referencing the Docker image page.

Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update live-migration.md

Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

---------

Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

atovpeko committed:
Timestamp vs timestamptz explanation (#3491)

Clarified the use of timestamp/timestamptz for partitioning hypertables

billy-the-fish committed:
fix: automatic compression compresses chunks automatically :-). (#3489)

swiffer committed:
Typo fix PotsgreSQL to PostgreSQL (#3495)

* PotsgreSQL to PostgreSQL

Signed-off-by: Matthias Wirtz <matthias.wirtz@hotmail.de>

VineethReddy02 committed:
Changelog for Oct 10th, 2024 (#3501)

* Changelog for Oct 10th, 2024

Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update about/changelog.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Update changelog.md

Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>

* Small edit for lists. 

Signed-off-by: Iain Cox <iain@timescale.com>

* Fix image

Signed-off-by: Iain Cox <iain@timescale.com>

---------

Signed-off-by: Vineeth Pothulapati <vineethpothulapati@outlook.com>
Signed-off-by: Iain Cox <iain@timescale.com>
Co-authored-by: Iain Cox <iain@timescale.com>

atovpeko committed:
Clarified compression policy parameters (#3494)

Co-authored-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
chore: remove section from the tiered storage troubleshooting. (#3503)

billy-the-fish committed:
fix: note about upgrades. (#3506)

billy-the-fish committed:
chore: update user management page. (#3485)

* chore: update user management page.

Signed-off-by: Iain Cox <iain@timescale.com>
Co-authored-by: atovpeko <114177030+atovpeko@users.noreply.github.com>

gayyappan committed:
Add documentation for disable_tiering (#3496)

* Add documentation for disable_tiering

Signed-off-by: Iain Cox <iain@timescale.com>
Co-authored-by: Yannis Roussos <iroussos@gmail.com>

atovpeko committed:
added the highlight (#3507)

Co-authored-by: Iain Cox <iain@timescale.com>

atovpeko committed:
Update the multi-ha docs (#3510)

* draft

solugebefola committed:
update: changelog for 10/18/2024 (#3514)

* update: changelog for 10/18/2024

* update: apply recommended changelog edits

VineethReddy02 committed:
Update Ingest data tree (#3505)

* update ingest data tree

Co-authored-by: Iain <iain@timescale.com>

atovpeko committed:
Fixed a broken link (#3520)

solugebefola committed:
Add hypertable creation wizard images to changelog.md (#3522)

* Add hypertable creation wizard images to changelog.md

Signed-off-by: Solar Olugebefola <12288273+solugebefola@users.noreply.github.com>

* Update changelog.md

Signed-off-by: Iain Cox <iain@timescale.com>

---------

Signed-off-by: Solar Olugebefola <12288273+solugebefola@users.noreply.github.com>
Signed-off-by: Iain Cox <iain@timescale.com>
Co-authored-by: Iain Cox <iain@timescale.com>

atovpeko committed:
Removed the unused redirect file (#3512)

atovpeko committed:
Added a note about downloading logs

atovpeko committed:
Added link to pricing (#3526)

billy-the-fish committed:
3545 docs rfc remove distributed hypertable example from the api ref (#3546)

* chore: remove distributed hypertable example.

atovpeko committed:
Add a page for foreign data wrappers (#3532)

* Add docs for FDW

---------

Co-authored-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
chore: explain performance improvements. (#3538)

* chore: explain performance improvements.

billy-the-fish committed:
chore: improved performance on deletes. (#3539)

* chore: improved performance on deletes.

Signed-off-by: Iain Cox <iain@timescale.com>
Co-authored-by: Yannis Roussos <iroussos@gmail.com>

billy-the-fish committed:
chore: replace placeholders in code example. (#3552)

* chore: replace placeholders in code example.

solugebefola committed:
Fix typo in quick-start main page (#3554)

Signed-off-by: Solar Olugebefola <12288273+solugebefola@users.noreply.github.com>

ramonguiu committed:
Changelog Oct 31 (#3555)

* Changelog Oct 31

Signed-off-by: Ramon Guiu <ramon@timescale.com>

* Update changelog.md

Signed-off-by: Ramon Guiu <ramon@timescale.com>

* Update changelog.md

Signed-off-by: Ramon Guiu <ramon@timescale.com>

* Apply suggestions from code review

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Ramon Guiu <ramon@timescale.com>

* Update changelog.md

Signed-off-by: Ramon Guiu <ramon@timescale.com>

* Update changelog.md

Signed-off-by: Ramon Guiu <ramon@timescale.com>

---------

Signed-off-by: Ramon Guiu <ramon@timescale.com>
Co-authored-by: Iain Cox <iain@timescale.com>

arajkumar committed:
Live-migration: Improve connection calculation (#3556)

Signed-off-by: Arunprasad Rajkumar <ar.arunprasad@gmail.com>

arajkumar committed:
Add resource recommendation for EC2 instance used in live-migration (#3561)

Signed-off-by: Arunprasad Rajkumar <ar.arunprasad@gmail.com>

VineethReddy02 committed:
Changelog Nov 08 (#3564)

billy-the-fish committed:
243 docs rfc add sql assistant to data mode doc (#3578)

* feat: first draft of sql assistant docs.

* feat: couple of review comments.

billy-the-fish committed:
fix: talk about pricing plans rather than tiers. (#3571)

billy-the-fish committed:
bug: track_io_timing is enabled by default (#3589)

billy-the-fish committed:
feat: remove old pgvectorizer page. (#3584)

* feat: remove old pgvectorizer page.
* bug: remove links to templates that have been removed from the langchain repo.

SrinidhiPRao committed:
Fix typo in Kubernetes installation documentation (#3574)

Co-authored-by: Iain Cox <iain@timescale.com>

Commit:
Add multi-node deprecation warning (#3547)

Signed-off-by: Iain Cox <iain@timescale.com>
Co-authored-by: Iain Cox <iain@timescale.com>

billy-the-fish committed:
feat: add information on multiple VPCs. (#3585)

* feat: add information on multiple VPCs.

Co-authored-by: atovpeko <114177030+atovpeko@users.noreply.github.com>

atovpeko committed:
Fixed links in the seo report (#3557)

Fixed broken links

philkra committed:
Remove support for PG12 after 2.12 (#3597)

rahilsondhi committed:
Changelog post for Nov 14, 2024 (#3591)

iroussos committed:
Update changelog with announcement on OSM 1.10 features (#3603)

* Update changelog with announcement on OSM 1.10 features

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Yannis Roussos <iroussos@gmail.com>

---------

Signed-off-by: Yannis Roussos <iroussos@gmail.com>
Co-authored-by: Iain Cox <iain@timescale.com>

atovpeko committed:
Fix broken links (#3604)

billy-the-fish committed:
chore: move the dimension_info argument definition into the functions that use it.

atovpeko committed:
IP Allow List (#3587)

Added a doc for IP allow list

billy-the-fish committed:
fix: remove variable from excerpt. (#3605)

atovpeko committed:
Add an intro about account/service/connection (#3490)

* Add an intro about account/service/connection
---------

Co-authored-by: Iain <iain@timescale.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/timescale/docs?pullRequest=3446) <br/>All committers have signed the CLA.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/timescale/docs?pullRequest=3446) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/timescale/docs?pullRequest=3446) before we can accept your contribution.<br/>**13** out of **14** committers have signed the CLA.<br/><br/>:white_check_mark: billy-the-fish<br/>:white_check_mark: swiffer<br/>:white_check_mark: atovpeko<br/>:white_check_mark: VineethReddy02<br/>:white_check_mark: solugebefola<br/>:white_check_mark: arajkumar<br/>:white_check_mark: ramonguiu<br/>:white_check_mark: gayyappan<br/>:white_check_mark: philkra<br/>:white_check_mark: rahilsondhi<br/>:white_check_mark: iroussos<br/>:white_check_mark: SrinidhiPRao<br/>:white_check_mark: jonatas<br/>:x: esclymeTS<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/timescale/docs?pullRequest=3446) it.</sub>

billy-the-fish commented:
The merge was a mess, replaced with https://github.com/timescale/docs/pull/3608.
